Sources/KituraContracts/CodableQuery/QueryEncoder.swift

```diff
@@ -199,7 +211,215 @@ public class QueryEncoder: Coder, Encoder, BodyEncoder {
+                encoder.anyDictionary[fieldName] = fieldValue
+            case .millisecondsSince1970:
+                let date = NSNumber(value: 1000 * fieldValue.timeIntervalSince1970)
+                encoder.dictionary[fieldName] = date.stringValue
+                encoder.anyDictionary[fieldName] = fieldValue
+            case .iso8601:
+                if #available(macOS 10.12, iOS 10.0, watchOS 3.0, tvOS 10.0, *) {
+                    encoder.dictionary[fieldName] = _iso8601Formatter.string(from: fieldValue)
+                    encoder.anyDictionary[fieldName] = fieldValue
+                } else {
+                    fatalError("ISO8601DateFormatter is unavailable on this platform.")
+                }
+            case .custom(let closure):
+                try closure(fieldValue, encoder)
+            default:
+                Log.warning("Unknown encoding strategy")
+            }
+        case let fieldValue as [Date]:
+            switch encoder.dateEncoder {
+            case .deferredToDate:
+                let dbs: [NSNumber] = fieldValue.map { NSNumber(value: $0.timeIntervalSinceReferenceDate) }
+                let strs: [String] = dbs.map { ($0).stringValue}
+                encoder.dictionary[fieldName] = strs.joined(separator: ",")
+                encoder.anyDictionary[fieldName] = fieldValue
+            case .secondsSince1970:
+                let dbs: [NSNumber] = fieldValue.map { NSNumber(value: $0.timeIntervalSince1970) }
+                let strs: [String] = dbs.map { ($0).stringValue}
+                encoder.dictionary[fieldName] = strs.joined(separator: ",")
+                encoder.anyDictionary[fieldName] = fieldValue
+            case .millisecondsSince1970:
+                let dbs: [NSNumber] = fieldValue.map { NSNumber(value: ($0.timeIntervalSince1970)/1000) }
+                let strs: [String] = dbs.map { ($0).stringValue}
+                encoder.dictionary[fieldName] = strs.joined(separator: ",")
+                encoder.anyDictionary[fieldName] = fieldValue
+            case .iso8601:
+                if #available(macOS 10.12, iOS 10.0, watchOS 3.0, tvOS 10.0, *) {
+                    let strs: [String] = fieldValue.map { _iso8601Formatter.string(from: $0) }
+                    encoder.dictionary[fieldName] = strs.joined(separator: ",")
+                    encoder.anyDictionary[fieldName] = fieldValue
+                } else {
+                    fatalError("ISO8601DateFormatter is unavailable on this platform.")
+                }
+            case .formatted(let formatter):
+                let strs: [String] = fieldValue.map { formatter.string(from: $0) }
+                encoder.dictionary[fieldName] = strs.joined(separator: ",")
+                encoder.anyDictionary[fieldName] = fieldValue
+            case .custom(let closure):
+                for element in fieldValue {
+                    try closure(element, encoder)
```

Comment:
It would be worth a comment that this calls us back with each serialized element individually, for the same `fieldName` key, and we build a comma-separated list (using the `=+=` operator defined below)